src/app/components/music-controls/music-controls.component.ts

```diff
@@ -0,0 +1,114 @@
+import { Component } from '@angular/core';
+import { AudioService } from '../../services/audio.service';
+import { StreamState, ITrack } from '../../types/interfaces';
+import { displayMillisecInMinSec } from '../../utils/utils';
+
+@Component({
+  selector: 'app-music-controls',
+  templateUrl: './music-controls.component.html',
+  styleUrls: ['./music-controls.component.scss']
+})
+export class MusicControlsComponent {
+  public state: StreamState;
+  public currentAudioID: string;
+  public currentTrackList: ITrack[];
+  public currentTrack: ITrack;
+  public isAutorenewed: boolean;
+  public isShuffled: boolean;
+  public isTracksListEnd: boolean = false;
+  public isVolumeOff: boolean = false;
+
+  constructor(
+    private audioService: AudioService
+  ) {
+    this.listenAudioService();
+    this.listenAudioChanges();
+    this.listenTrackListChanges();
+    this.isAutorenewed = this.audioService.getAutorenew();
+    this.isShuffled = this.audioService.getShuffle();
+  }
+
+  listenAudioService() {
+    this.audioService.getState().subscribe(newState => {
+      this.state = newState;
+      if (newState.volume > 0) {
+        this.isVolumeOff = false;
+      } else {
+        this.isVolumeOff = true;
+      }
+    })
+  }
+
+  listenAudioChanges() {
+    this.audioService.getAudioIDChange().subscribe(newAudioID => {
+      this.currentAudioID = newAudioID;
+      this.currentTrackList = this.audioService.getTrackList();
+      this.getAudioDetails();
+    })
+  }
+
+  listenTrackListChanges() {
+    this.audioService.getTrackListChange().subscribe((newTrackList: ITrack[]) => {
+      this.currentTrackList = newTrackList;
+      this.getAudioDetails();
+    })
+  }
+
+  getAudioDetails() {
+    this.currentTrack = this.currentTrackList.find(track => track.id === this.currentAudioID);
+    if (this.currentTrack) {
+      const isLastAudio = this.currentTrack.track_number + 1 === this.currentTrackList.length;
+      this.isTracksListEnd = isLastAudio;
+    }
+  }
+
+  convertTime(time: number) {
+    return displayMillisecInMinSec(time);
+  }
+
+  playPause() {
+    if (this.state.playing) {
+      this.pause()
+    } else {
+      this.play()
+    }
+  }
+
+  play() {
+    this.audioService.play()
```

Comment:
if you don't do anything else instead of just calling service method - just move this call into a general method

```
    if (this.state.playing) {
      this.audioService.pause();
    } else {
      this.audioService.play();
    }
```